src/pages/signup/index.tsx

```diff
@@ -1,5 +1,225 @@
-const Signup = () => (
-  <div>SignUp Page</div>
-);
+/* eslint-disable no-unused-vars */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import React, { useState } from 'react';
+import {
+  Button, CssBaseline, TextField, Paper, Box, Grid, Typography, RadioGroup, FormControlLabel, Radio,
+} from '@mui/material';
+import { Link } from 'react-router-dom';
+import { useFormik } from 'formik';
+import validationSchema from './schema';
+import imageSrc from '../../assets/loginImg.jpg';
+import {
+  boxStyle, textFieldStyle, buttonStyle, gridStyle, imageStyle, fileUploadStyle,
+} from './classes';
+import './style.css';
+
+const handleFileUpload = (_event: React.ChangeEvent<HTMLInputElement>) => {
+  // eslint-disable-next-line no-console
+  console.log('File uploaded successfully.');
+};
+
+const Signup = () => {
+  const [userType, setUserType] = useState('');
+  const [showAdditionalFields, setShowAdditionalFields] = useState(false);
+
+  const handleUserTypeChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setUserType(event.target.value);
+    setShowAdditionalFields(event.target.value === 'therapist');
+  };
+
+  const formik = useFormik({
+    initialValues: {
+      username: 'yasser',
+      email: 'yasser@example.com',
+      password: '123456789',
+      confirmPassword: '123456789',
+      major: 'Psychology',
+      hourlyRate: '100',
+    },
+    validationSchema,
+    onSubmit: (values) => {
+      // eslint-disable-next-line no-console
+      console.log(JSON.stringify(values, null, 2));
+    },
+  });
+
+  const renderAdditionalFields = () => {
+    if (showAdditionalFields) {
+      return (
+        <>
+          <TextField
+            margin="normal"
+            required
+            autoFocus
+            sx={textFieldStyle}
+            id="major"
+            name="major"
+            label="major"
+            type="major"
+            onChange={formik.handleChange}
+            error={formik.touched.major && Boolean(formik.errors.major)}
+            helperText={formik.touched.major && formik.errors.major}
+          />
+          <TextField
+            margin="normal"
+            required
+            autoFocus
+            sx={textFieldStyle}
+            id="hourlyRate"
+            name="hourlyRate"
+            label="hourlyRate"
+            type="hourlyRate"
```

Comment:
input type is not associated with it's name, Such common input types (email, name, number, password)